### ISSUE TITLE
"Factory" method has a problem

### ISSUE THREAD

**Issue by beebapcay** [GeekyAnts/NativeBase#3901]
**Describe the bug**
I create my own component. Then wrap it in a Factory method to be able to use style props. But it seems incompatible with some Components that need to pass more data props to FlatList

**To Reproduce**
1. Create a Component containing a FlatList, pass it: dataList
2. Then wrap it in Factory(`Your Component`)
3. When using this Component and passing it to a dataList, rendering will not be possible

**Expected behaviour**
Factory can be used to wrap self-defined Components

**Platform**
 - NativeBase version: 3.0.6
 - Device: Samsung S9+
 - OS: Android
 - Expo sdk version: 42

**Comment by MD-REHMAN** [comment]
Thanks @beebapcay for reporting. We'll push the fix in the upcoming patch.

**Comment by vxm5091** [comment]
I'm still having a similar issue. I'm able to render the components themselves but passing down style props like `my='6'` doesn't work. Am I doing something wrong? 

```
export const Grid = (props: GridProps) => {
  const { data } = props;
  
  const FactorySquare = Factory(Square)

  return (
    <FlatList
      w='100%'
      columnWrapperStyle={{
        justifyContent: 'space-between'
      }}
      data={data}
      horizontal={false}
      numColumns={3}
      renderItem={({ item }) => (
        <FactorySquare
          buddySquareLandingData={item}
          key={item.id}
          my={6}
        />
          
      )}
    />
  );
};
```

**Comment by makkarMeenu** [comment]
It has been fixed in NativeBase 3.2.2 .
